pandasai/helpers/anonymizer.py

```diff
@@ -175,5 +178,12 @@ def anonymize_dataframe_head(
                 pd.eval(cell_value) if cell_value in ["True", "False"] else cell_value
             )
     # restore the original data types
-    data_frame = data_frame.astype(dtypes)
+    for i in range(len(data_frame.columns)):
```

Comment:
Could you add a comment to make it more understandable? This anonymizer is growing in complexity, we'd better have it well specced!

Reply:
ok, I will add that

Reply:
I have added the comments